File: frontend/src/types/repository.ts
import { Issue } from './issue';

type Repository = {
  id: number;
  name: string;
  url: string;
  description: string;
  starCount: number;
  forkCount: number;
  openIssueCount: number;
  topics: string;
  license: string;
  language: string;
  issues: Issue[];
};

export type { Repository };

File: backend/lib/model/repository.go
package model

import "time"

type (
	Repository struct {
		RepositoryID     int64     `bson:"repository_id"`
		Name             string    `bson:"name"`
		URL              string    `bson:"url"`
		Description      string    `bson:"description"`
		StarCount        *int      `bson:"star_count"`
		ForkCount        *int      `bson:"fork_count"`
		OpenIssueCount   *int      `bson:"open_issue_count"`
		Topics           []string  `bson:"topics"`
		License          string    `bson:"license"`
		Language         string    `bson:"language"`
		UpdatedAt        time.Time `bson:"updated_at"`
		IssueInitialized bool      `bson:"issue_initialized"`
		GitHubCreatedAt  time.Time `bson:"github_created_at"`
		GitHubUpdatedAt  time.Time `bson:"github_updated_at"`
		Issues           []*Issue  `bson:"issues"`
	}
)
